internal/status/status.go

```diff
@@ -109,7 +110,7 @@ func (s *Status) WithDetails(details ...proto.Message) (*Status, error) {
 	// s.Code() != OK implies that s.Proto() != nil.
 	p := s.Proto()
 	for _, detail := range details {
-		any, err := ptypes.MarshalAny(detail)
+		any, err := anypb.New(detail)
```

Comment:
Just curious, do we not require a conversion to the V2 message here?

Reply:
Because I imported the google.golang.org/protobuf/proto library instead of github.com/golang/protobuf/proto library, if you think it's wrong, I can take the old import and convert accordingly.